src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -370,4 +376,36 @@ export class TerminalPanel extends Panel {
 		}
 		return path;
 	}
+
+	/**
+	 * Surrounds a Windows path with quotation marks if necessary
+	 */
+	public static prepareWindowsPathForTerminal(path: string): string {
+		if (/\s+/.test(path)) {
+			return `"${path}"`;
+		}
+		return path;
+	}
+
+	/**
+	 * Adds quotes to a path if it contains whitespaces, and formats it for the correct windows subshell.
+	 * It is assumed there is an active terminal instance.
+	 */
+	private _preparePathForTerminal(path: string): string {
+		if (platform.isWindows) {
+			const terminal = this._terminalService.getActiveInstance();
+			const shell = terminal.getShellName();
+			for (const [format, formatter, isWinPath] of WIN_PATH_FORMATTERS) {
```

Comment:
Let's use `WIN_PATH_FORMATTERS.forEach(` to keep it consistent with the rest of the code.

Reply:
Actually never mind. The regular expression for WSL bash is becoming quite long, so I may as well use an interface to split up the line.

Reply:
I don't think using `forEach` will work because I need to break out of the loop. Either I would need to nest an `if` inside of it which would get ugly or use some other method (such as `Array.prototype.find`, but it seems the typescript target is set to es5 so that's not an option).